### PR TITLE
HPCC-25686 Thor crash in FileIO->close()

### PR DESCRIPTION
**Pull request by mckellyln**
Signed-off-by: Mark Kelly <mark.kelly@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

smoke test

### CODE REVIEW

**Comment by mckellyln** [comment]
@jakesmith do you also think we should throw if we cannot open the file ?
As in -
```
    Owned<IFileIO> partOutputIO = createMultipleWrite(this, *partDesc, diskRowMinSz, twFlags, compress, ecomp, this, &abortSoon, (external&&!query) ? &tempExternalName : NULL);
    if (!partOutputIO)
        throw MakeActivityException(activity, ...)
```

**Comment by jakesmith** [comment]
> @jakesmith do you also think we should throw if we cannot open the file ?
> As in -
> 
> ```
>     Owned<IFileIO> partOutputIO = createMultipleWrite(this, *partDesc, diskRowMinSz, twFlags, compress, ecomp, this, &abortSoon, (external&&!query) ? &tempExternalName : NULL);
>     if (!partOutputIO)
>         throw MakeActivityException(activity, ...)
> ```

@mckellyln - it already throws an exception if fails to open (see line 802)

File: thorlcr/activities/thdiskbaseslave.cpp [hunk 1]
```diff
@@ -433,14 +433,17 @@ void CDiskWriteSlaveActivityBase::close()
             outraw.clear();
         }
 
-        Owned<IFileIO> tmpFileIO;
+        if (outputIO)
```
Comment (jakesmith) [inline thorlcr/activities/thdiskbaseslave.cpp:436]: looking again, if outputIO is unset, it implies neither 'out' or 'outraw' can be set either.

It won't make any practical difference, but clearer if this conditional block includes lines 417-434 too.

**Review by jakesmith — COMMENTED** [review]
Not imperative, but one improvement suggestion.

File: thorlcr/activities/thdiskbaseslave.cpp (hunk 1 above)
Comment (mckellyln) [reply thorlcr/activities/thdiskbaseslave.cpp:436]: ok, I will move it there.  thx

**Comment by mckellyln** [comment]
@jakesmith pushed update.  If ok I will squash

**Comment by mckellyln** [comment]
Squashed

**Review by jakesmith — APPROVED** [review]
Looks good.

**Pull request merged**